Show hint that PHP 5.6 will not be supported in Nextcloud 14 anymore

MorrisJobke opened the pull request:
Looks then like this:

![bildschirmfoto 2017-12-04 um 22 51 42](https://user-images.githubusercontent.com/245432/33578156-cb55d420-d945-11e7-80c8-7fd084882776.png)

Fixes #7386 

* Feedback on the wording is very welcome.
* Should it be red or black (warning vs info)?
* Should it be combined with the other version info that is already present or is this too little?

MorrisJobke committed:
Show hint that PHP 5.6 will not be supported in Nextcloud 14 anymore

Signed-off-by: Morris Jobke <hey@morrisjobke.de>

ChristophWurst commented on core/js/setupchecks.js:
```diff
@@ -122,6 +122,12 @@
 					}
+					if(data.phpSupported && data.phpSupported.version.substr(0, 3) === '5.6') {
+						messages.push({
+							msg: t('core', 'You are currently running PHP 5.6. The current major version of Nextcloud - version 13 - will be the last that is supported on PHP 5.6. It is recommended to upgrade the PHP version to be able to upgrade to the upcoming Nextcloud 14.', {version: data.phpSupported.version}),
```
'upcoming' sounds like it's about to be released. How about '…  be able to upgrade to Nextcloud 14'?

rullzer replied:
I would also link here to the page that is plannend on nextcloud.com

tflidd replied:
The current major version ... is the last version supporting PHP 5.6.
For NC13 it will be a fact, but once it is out it is a fact, so we don't need the more complicated "will be"-structure. And with "supporting", it is a bit shorter. Keep it simple.

MorrisJobke replied:
better now?

blizzz commented:
Also, perhaps red is too much for this. People with either thing something is fundamentally wrong, or worse, get the feeling that such messages are not really important. Yellow/light orange might be better suited as a soft heads up.

MorrisJobke commented:
> Yellow/light orange might be better suited as a soft heads up.

We don't have this yet - there is only red and black. Then I would go for black, which should be fine as well.

ChristophWurst commented on core/js/setupchecks.js:
```diff
@@ -122,6 +122,12 @@
 					}
+					if(data.phpSupported && data.phpSupported.version.substr(0, 3) === '5.6') {
+						messages.push({
+							msg: t('core', 'You are currently running PHP 5.6. The current major version of Nextcloud is the last that is supported on PHP 5.6. It is recommended to upgrade the PHP version to be able to upgrade to Nextcloud 14.', {version: data.phpSupported.version}),
```
the `version` parameter is unused now.

juliusstoerrle commented:
I think it might be a good Idea to recommend a Version (7.1) maybe even 7.2 which people should upgrade to, or at least say what will be the next minimum requirement.

MorrisJobke commented:
> I think it might be a good Idea to recommend a Version (7.1) maybe even 7.2 which people should upgrade to, or at least say what will be the next minimum requirement.

The recommendation is PHP 7.0+, because we have not seen a big difference here. Often people can not easily choose the PHP version and should stick to what the distribution offers.

MorrisJobke commented:
> I think it might be a good Idea to recommend a Version (7.1) maybe even 7.2 which people should upgrade to, or at least say what will be the next minimum requirement.

I added this and fixed the comment by @ChristophWurst.

This is ready for review now.

LukasReschke reviewed (approved):
🐘
